src/v/cloud_storage/async_manifest_view.h

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cloud_storage/fwd.h"
+#include "cloud_storage/materialized_segments.h"
+#include "cloud_storage/partition_manifest.h"
+#include "cloud_storage/remote.h"
+#include "cloud_storage/spillover_manifest.h"
+#include "cloud_storage/types.h"
+#include "cloud_storage_clients/types.h"
+#include "model/metadata.h"
+#include "model/timestamp.h"
+#include "utils/intrusive_list_helpers.h"
+#include "utils/retry_chain_node.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/core/timed_out_error.hh>
+
+#include <algorithm>
+#include <exception>
+#include <map>
+
+namespace cloud_storage {
+
+/// Materialized spillover manifest
+///
+/// The object contains the manifest, semaphore units,
+/// and it is also a part of an intrusive list.
+struct materialized_manifest
+  : ss::enable_shared_from_this<materialized_manifest> {
+    materialized_manifest(
+      const model::ntp& ntp,
+      model::initial_revision_id rev,
+      ss::semaphore_units<> u)
+      : manifest(ntp, rev)
+      , _units(std::move(u)) {}
+
+    materialized_manifest(spillover_manifest manifest, ss::semaphore_units<> u)
+      : manifest(std::move(manifest))
+      , _units(std::move(u)) {}
+
+    materialized_manifest() = delete;
+    ~materialized_manifest() = default;
+    materialized_manifest(const materialized_manifest&) = delete;
+    materialized_manifest(materialized_manifest&&) = delete;
+    materialized_manifest& operator=(const materialized_manifest&) = delete;
+    materialized_manifest& operator=(materialized_manifest&&) = delete;
+
+    spillover_manifest manifest;
+    ss::semaphore_units<> _units;
+    intrusive_list_hook _hook;
+};
+
+/// Collection of materialized manifests
+/// The cache stores manifests in memory (hence "materialized").
+/// The memory limit is specified in bytes. The cache uses LRU
+/// eviction policy and will try to evict least recently used
+/// materialized manifest.
+class materialized_manifest_cache {
```

Comment:
This file should be called materialized_manifest_cache.h, right?  The current name doesn't seem to match the contents.

Reply:
This is only the first part of the PR and the second one has a `async_manifest_view` class in it that uses the cache.